lib/options.nix

```
@@ -403,5 +403,41 @@ rec {
         if cfg ? lazyLoad then lib.literalMD "`false` when lazy-loading is enabled." else true;
       example = false;
     };
+
+  /**
+    Create an option for an executable currently unpackaged in nixpkgs.
+    This option will throw an error if a value is not explicitly set by the end user
+  */
+  mkUnpackagedOption =
+    fullName: name:
+    lib.mkOption {
+      type = types.nullOr types.package;
+      description = ''
+        Package to use for ${name}.
+        Nixpkgs does not include this package, and as such an external derivation or null must be provided.
+      '';
+      default = throw ''
+        Nixvim (${fullName}): No package is known for ${name}, to resolve this either:
+          - install externally and set this option to `null`
+          - or provide a derviation to install this package
+      '';
+      defaultText = lib.literalExpression "No package";
```

Nit: 
```suggestion
      defaultText = lib.literalMD "No package";
```

Non-code defaults are usually better as markdown than expression.

Could even say something like `"throws when undefined"`?